cpu/sam0_common/periph/uart.c

```diff
@@ -396,6 +423,17 @@ static inline void irq_handler(unsigned uartnum)
     }
 #endif
 
+#ifdef MODULE_PERIPH_UART_RX_START
+    if (status & SERCOM_USART_INTFLAG_RXS && dev(uartnum)->INTENSET.bit.RXS) {
+        /* clear receive start flag */
+        dev(uartnum)->INTFLAG.reg = SERCOM_USART_INTFLAG_RXS;
```

Comment:
Can this become a race like in #16478 if `dev(uartnum)->INTENSET.bit.RXS` has been cleared between raising the IRQ and handling the IRQ?

Reply:
I don't see how this could cause an issue.

We could of course always clear all flags, but for the collision detection feature it would be handy if the collision bit remains untouched by the interrupt handler and we can simply read it out after `uart_write()`

Reply:
Okay, I just found the fallowing paragraph in the SAML21 datasheet:

> An interrupt request is generated when the interrupt flag is set and if the corresponding interrupt is enabled. The
> interrupt request remains active until either the interrupt flag is cleared, the interrupt is disabled, or the USART is
> reset.

Disabling the IRQ will stop triggering the ISR over and over again (which wasn't the case for the STM32).

> We could of course always clear all flags, but for the collision detection feature it would be handy if the collision bit remains untouched by the interrupt handler and we can simply read it out after uart_write()

How will this improve the collision detection? I think we can't distinguish between an foreign or our start bit. The TXSTART IRQ would be triggered either way.

Reply:
Collision detection is a separate feature of the UART unrelated to this PR (`COLDEN` / `STATUS.COLL`)

Reply:
I'm silly, this already gets cleared on line 419